src/transport_performance/gtfs/report/report_utils.py

```diff
@@ -0,0 +1,183 @@
+        "parent_station",
+        "platform_code",
+    ],
+    "calendar_dates": [],
+    "calendar": [],
+    "trips": [
+        "trip_headsign",
+        "block_id",
+        "shape_id",
+        "wheelchair_accessible",
+    ],
+    "shapes": [],
+}
+
+
+class TemplateHTML:
+    """A class for inserting HTML string into a docstring.
+
+    Attributes
+    ----------
+    template : str
+        A string containing the HTML template.
+
+    Methods
+    -------
+    insert(placeholder: str, value: str, replace_multiple: bool = False)
+        Insert values into the HTML template
+    get_template()
+        Returns the template attribute
+
+    """
+
+    def __init__(self, path: Union[str, pathlib.Path]) -> None:
+        """Initialise the TemplateHTML object.
+
+        Parameters
+        ----------
+        path : Union[str, pathlib.Path]
+            The file path of the html template
+
+        Returns
+        -------
+        None
+
+        """
+        _handle_path_like(path, "path")
+        with open(path, "r", encoding="utf8") as f:
+            self.template = f.read()
+        return None
+
+    def insert(
+        self, placeholder: str, value: str, replace_multiple: bool = False
+    ) -> None:
+        """Insert values into the html template.
+
+        Parameters
+        ----------
+        placeholder : str
+            The placeholder name in the template. This is a string. In the
+            template it should be surrounded by square brackets.
+        value : str
+            The value to place in the placeholder
+            location.
+        replace_multiple : bool, optional
+            Whether or not to replace multiple placeholders that share the same
+            placeholder value, by default False
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValueError
+            A ValueError is raised if there are multiple instances of a
+            place-holder but 'replace_multiple' is not True
+
+        """
+        _type_defence(placeholder, "placeholder", str)
+        _type_defence(value, "value", str)
+        _type_defence(replace_multiple, "replace_multiple", bool)
+        occurences = len(self.template.split(f"[{placeholder}]")) - 1
+        if occurences > 1 and not replace_multiple:
+            raise ValueError(
+                "You have selected not to replace multiple"
+                "placeholders of the same value, however"
+                "placeholders occur more than once. \n"
+                "If you would like to allow this, set the"
+                "replace_multiple param to True"
+            )
+
+        self.template = self.template.replace(f"[{placeholder}]", value)
+
+    def get_template(self) -> str:
+        """Get the template attribute of the TemplateHTML object.
+
+        Returns
+        -------
+        str
+            The template attribute
+
+        """
+        return self.template
+
+
+def set_up_report_dir(
```

Comment:
Please can you debug this:

```
set_up_report_dir("TESTING", overwrite=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 33, in set_up_report_dir
FileNotFoundError: [Errno 2] No such file or directory: 'TESTING/gtfs_report'
>>> set_up_report_dir("TESTING/TESTING1", overwrite=True)
Creating parent directory: /Users/richardleyshon/Documents/transport-network-performance/TESTING
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 33, in set_up_report_dir
FileNotFoundError: [Errno 2] No such file or directory: 'TESTING/TESTING1/gtfs_report' 

```

It looks like `_check_parent_dir_exists` doesn't have the behaviour you need here. Needs adapting.

Reply:
The issue here was that the `path` param was being passed to `_check_parent_dir_exists()`, which would only create the directory up to `testing/testing` in the case that `path=testing/testing/testing` as it is only trying to create the 'parent' directory. Functionality has been updated to reflect this.